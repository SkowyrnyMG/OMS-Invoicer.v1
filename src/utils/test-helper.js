import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
