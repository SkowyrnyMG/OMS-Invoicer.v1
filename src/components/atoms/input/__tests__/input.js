import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import Input from '../input';

describe('Input', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<Input />);
  });
});
