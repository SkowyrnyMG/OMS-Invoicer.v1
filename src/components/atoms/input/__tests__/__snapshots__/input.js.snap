// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should display in the document with correct styles 1`] = `
.c0 {
  margin: 1rem 0;
  padding: 0.5rem;
  width: 100%;
  height: 3rem;
  font-size: 16px;
  color: #8F8F8F;
  border: none;
  outline: none;
  border-bottom: 2px solid currentColor;
  -webkit-transition: 0.5s -webkit-transform;
  -webkit-transition: 0.5s transform;
  transition: 0.5s transform;
}

.c0:focus {
  -webkit-transform: translateX(2px);
  -ms-transform: translateX(2px);
  transform: translateX(2px);
}

.c0 input,
.c0 input::-webkit-input-placeholder {
  color: currentColor;
}

.c0 input,
.c0 input::-moz-placeholder {
  color: currentColor;
}

.c0 input,
.c0 input:-ms-input-placeholder {
  color: currentColor;
}

.c0 input,
.c0 input::placeholder {
  color: currentColor;
}

<input
  class="c0"
/>
`;
